Code/GraphMol/RGroupDecomposition/catch_rgd.cpp

```diff
@@ -808,3 +808,175 @@ TEST_CASE("Mol matches core") {
+    0.2198    1.6246    0.0000 R#  0  0  0  0  0  1  0  0  0  0  0  0
+   -0.6493    0.1262    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -1.5142    0.6280    0.0000 R#  0  0  0  0  0  1  0  0  0  0  0  0
+   -0.6513   -0.8736    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+    0.2137   -1.3754    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  2  0
+  2  3  1  0
+  3  4  1  0
+  3  5  2  0
+  5  6  1  0
+  5  7  1  0
+  7  8  2  0
+  8  1  1  0
+M  RGP  2   4   2   6   1
+M  END
+)CTAB"_ctab;
+    core->removeConformer(0);
+    CHECK(MolToCXSmiles(*core, p) ==
+          "c1cc([2*])c([1*])cn1 |atomProp:3.dummyLabel.R2:5.dummyLabel.R1|");
+    relabelMappedDummies(*core);
+    CHECK(MolToCXSmiles(*core, p) ==
+          "c1cc(*)c(*)cn1 |atomProp:3.dummyLabel.R2:5.dummyLabel.R1|");
+  }
+  SECTION("AtomMap and Isotope in, MDLRGroup out - AtomMap has priority") {
+    auto core = "c1cc([4*:2])c([3*:1])cn1"_smiles;
+    CHECK(
+        MolToCXSmiles(*core, p) ==
+        "c1cc([4*:2])c([3*:1])cn1 |atomProp:3.dummyLabel.*:3.molAtomMapNumber.2:5.dummyLabel.*:5.molAtomMapNumber.1|");
+    relabelMappedDummies(*core);
+    CHECK(MolToCXSmiles(*core, p) ==
+          "c1cc(*)c(*)cn1 |atomProp:3.dummyLabel.R2:5.dummyLabel.R1|");
+  }
+  SECTION("AtomMap and Isotope in, MDLRGroup out - force Isotope priority") {
+    auto core = "c1cc([4*:2])c([3*:1])cn1"_smiles;
+    CHECK(
+        MolToCXSmiles(*core, p) ==
+        "c1cc([4*:2])c([3*:1])cn1 |atomProp:3.dummyLabel.*:3.molAtomMapNumber.2:5.dummyLabel.*:5.molAtomMapNumber.1|");
+    relabelMappedDummies(*core, Isotope);
+    CHECK(MolToCXSmiles(*core, p) ==
+          "c1cc(*)c(*)cn1 |atomProp:3.dummyLabel.R4:5.dummyLabel.R3|");
+  }
+  SECTION(
+      "AtomMap, Isotope and MDLRGroup in, MDLRGroup out - AtomMap has priority") {
+    auto core = ROMOL_SPTR(new ROMol(*allDifferentCore));
+    relabelMappedDummies(*core);
+    CHECK(MolToCXSmiles(*core, p) ==
+          "c1cc(*)c(*)cn1 |atomProp:3.dummyLabel.R4:5.dummyLabel.R3|");
+  }
+  SECTION(
+      "AtomMap, Isotope and MDLRGroup in, MDLRGroup out - force Isotope priority") {
+    auto core = ROMOL_SPTR(new ROMol(*allDifferentCore));
+    relabelMappedDummies(*core, Isotope);
+    CHECK(MolToCXSmiles(*core, p) ==
+          "c1cc(*)c(*)cn1 |atomProp:3.dummyLabel.R6:5.dummyLabel.R5|");
+  }
+  SECTION(
+      "AtomMap, Isotope and MDLRGroup in, MDLRGroup out - force MDLRGroup priority") {
+    auto core = ROMOL_SPTR(new ROMol(*allDifferentCore));
+    relabelMappedDummies(*core, MDLRGroup);
+    CHECK(MolToCXSmiles(*core, p) ==
+          "c1cc(*)c(*)cn1 |atomProp:3.dummyLabel.R2:5.dummyLabel.R1|");
+  }
+  SECTION(
+      "AtomMap, Isotope and MDLRGroup in, AtomMap out - AtomMap has priority") {
+    auto core = ROMOL_SPTR(new ROMol(*allDifferentCore));
+    relabelMappedDummies(
+        *core, static_cast<RGroupLabelling>(AtomMap | Isotope | MDLRGroup),
+        AtomMap);
+    CHECK(
+        MolToCXSmiles(*core, p) ==
```

Comment:
I think the dummy Labels shouldn't still be in the output here?

Reply:
This applies below a few times as well

Reply:
Ah, good point. Fixed.